db/app/controllers.js

```diff
@@ -15,7 +17,33 @@ const getArticles = (req, res) => {
   });
 };
 
+const getArticleByID = (req, res, next) => {
+  const id = req.params.id;
+  return selectArticleByID(id)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch(next);
+};
+
+const postComment = (req, res, next) => {
+
+  const id = req.params.id;
+  const { body } = req;
+
+  // Making sure the article exists
+  selectArticleByID(id).catch(next);
```

Comment:
This check isn't necessary as if the id doesn't exist then PSQL will throw an error from the model.